include/geopackage/WKB.hpp

```diff
@@ -0,0 +1,149 @@
+
+    /**
+     * Read a WKB multipoint into a cartesian model.
+     * @return multipoint_t 
+     */
+    static multipoint_t read_multipoint(const byte_vector&, int&, uint8_t);
+
+    /**
+     * Read a WKB multilinestring into a cartesian model.
+     * @return multilinestring_t 
+     */
+    static multilinestring_t read_multilinestring(const byte_vector&, int&, uint8_t);
+
+    /**
+     * Read a WKB multipolygon into a cartesian model.
+     * @return multipolygon_t 
+     */
+    static multipolygon_t read_multipolygon(const byte_vector&, int&, uint8_t);
+};
+
+/**
+ * EPSG 5070 projection definition for use with boost::geometry.
+ * 
+ * @note this is required because boost 1.72.0 does not
+ *       have an EPSG definition for 5070 in boost::srs::epsg.
+ */
+const auto epsg5070 = bg::srs::dpar::parameters<>(bg::srs::dpar::proj_aea)
+                                                 (bg::srs::dpar::ellps_grs80)
+                                                 (bg::srs::dpar::towgs84, {0,0,0,0,0,0,0})
+                                                 (bg::srs::dpar::lat_0, 23)
+                                                 (bg::srs::dpar::lon_0, -96)
+                                                 (bg::srs::dpar::lat_1, 29.5)
+                                                 (bg::srs::dpar::lat_2, 45.5)
+                                                 (bg::srs::dpar::x_0, 0)
+                                                 (bg::srs::dpar::y_0, 0);
+
+/**
+ * EPSG 3857 projection definition for use with boost::geometry.
+ * 
+ * @note this is required because boost 1.72.0 does not
+ *       have an EPSG definition for 3857 in boost::srs::epsg.
+ */
+const auto epsg3857 = bg::srs::dpar::parameters<>(bg::srs::dpar::proj_merc)
+                                                 (bg::srs::dpar::units_m)
+                                                 (bg::srs::dpar::no_defs)
+                                                 (bg::srs::dpar::a, 6378137)
+                                                 (bg::srs::dpar::b, 6378137)
+                                                 (bg::srs::dpar::lat_ts, 0)
+                                                 (bg::srs::dpar::lon_0, 0)
+                                                 (bg::srs::dpar::x_0, 0)
+                                                 (bg::srs::dpar::y_0, 0)
+                                                 (bg::srs::dpar::k, 1);
+
+struct wkb::wgs84 : public boost::static_visitor<geojson::geometry>
+{
+    wgs84(uint32_t srs, const bg::srs::transformation<>& tr)
+        : srs(srs)
+        , tr(tr) {};
+
+    geojson::geometry operator()(point_t& g);
+    geojson::geometry operator()(linestring_t& g);
+    geojson::geometry operator()(polygon_t& g);
+    geojson::geometry operator()(multipoint_t& g);
+    geojson::geometry operator()(multilinestring_t& g);
+    geojson::geometry operator()(multipolygon_t& g);
```

Comment:
Given how they're implemented, the really idiomatic thing to do would be to make these all `virtual`, and have a subclass for the `srs == 4326` case. But, that's probably not actually worth doing here and now.

Reply:
We can postulate this in an issue as a future enhancement if we don't want to refactor it on this PR.

Reply:
Let's leave it alone, and consider it if we need to account for more than the one or two cases we have now.

Reply:
This might also factor into #567 somewhat due to the transformation calls.

I think in general we will only have the 2 cases: (1) geometry is in EPSG:4326, and (2) geometry is not in EPSG:4326.

Though, this brings up another point of if we want to work in EPSG:4326 in-memory, or transform to a numerically better projection for our expected domain (i.e. EPSG:5070 for CONUS). There's probably a lot more to consider with that though, so I agree making an issue is the better approach for this.